orca-qos/src/main/kotlin/com/netflix/spinnaker/orca/qos/bufferstate/ActiveExecutionsBufferStateSupplier.kt

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2018 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.orca.qos.bufferstate
+
+import com.netflix.spectator.api.Registry
+import com.netflix.spinnaker.orca.qos.BufferState
+import com.netflix.spinnaker.orca.qos.BufferState.ACTIVE
+import com.netflix.spinnaker.orca.qos.BufferState.INACTIVE
+import com.netflix.spinnaker.orca.qos.BufferStateSupplier
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.scheduling.annotation.Scheduled
+import org.springframework.stereotype.Component
+
+/**
+ * Supplies the buffering state based on a threshold of system-wide active executions.
+ *
+ * As an immediate implementation, this supplier depends on [RedisActiveExecutionMonitor] to update the registry with
+ * system-wide active executions. The polling interval of that monitor will determine the resolution of the buffering
+ * state. By default, this is 60 seconds.
+ */
+@Component
+class ActiveExecutionsBufferStateSupplier(
```

Comment:
That's exactly the behavior of this PR. I don't want to have the logic of determining whether executions should potentially be buffered (whatever it may be in the future) in the actual persist execution path, since it could potentially be an expensive operation.

Reply:
Oh, so _all_ executions get buffered when buffering is happening? Then the decider will gate some of them thru quickly?

Reply:
That's right.

Reply:
Ok, that seems legit. I like the simplicity.